lib/nostrum/voice.ex

```diff
@@ -474,6 +474,29 @@ defmodule Nostrum.Voice do
     end
   end
 
+  @doc """
+  Listen for incoming voice RTP packets.
+
+  ## Parameters
+    - `guild_id` - ID of guild that the bot is listening to.
+    - `num_packets` - Number of packets to wait for.
+
+  Returns a list of 2-element tuples in the form `{rtp_header, opus_frame}`.
+
+  This function will block until the specified number of packets is received.
+  """
+  @doc since: "0.5.0"
+  @spec listen(Guild.id(), pos_integer) :: [{binary, binary}] | {:error, String.t()}
+  def listen(guild_id, num_packets) do
+    voice = get_voice(guild_id)
+
+    if VoiceState.ready_for_rtp?(voice) do
+      Enum.map(1..num_packets, fn _ -> Audio.get_rtp_packet(voice) end)
+    else
+      {:error, "Must be connected to voice channel to listen for incoming data."}
```

Comment:
Since we're in a library here, do you think it would make more sense to return an atom as the error, such as

```suggestion
      {:error, :not_connected}
```

I think it would make it easier for users to match on the error, should this function start returning other errors.

Reply:
I kept it a string because the rest of the voice API returns a descriptive error message and wanted to conform to that without breaking the rest. Maybe the rest should be changed but could be breaking

Reply:
Perfectly reasonable. Thanks!